sw/device/tests/crypto/cryptotest/firmware/hmac.c

```diff
@@ -0,0 +1,132 @@
+    0xB4A8B698, 0x1986B902, 0x7251045B, 0x3C827C6F, 0x00909D12, 0x81ABC8F9,
+    0x62F2FCB6, 0x15B63124, 0x66F60052, 0xAD637669, 0x522779CF, 0x07E9FBA8,
+    0x1258E541, 0x860719EF, 0x1D4F5386, 0xA9B04F7C, 0x6E98A861, 0xEFADEBA6,
+    0x900E1EC8, 0xB290DBCE, 0x05946814, 0xB83A01CE, 0x4EEC86BD, 0xAE836C6C,
+    0x20182AAE, 0x4476F6F4, 0x7C4A0A31, 0x7D2809BA, 0x367B29B9, 0x42444BEA,
+    0xDFD6025C, 0x1E665207, 0x18E0895B, 0x20D435DB, 0xC509A6D6, 0x8CC19AB1,
+    0xA5D39BD2, 0xAB479AD5, 0x5786D029, 0x2E4B7CD7, 0xB77A3D76, 0xE2A09962,
+};
+
+status_t handle_hmac(ujson_t *uj) {
+  // Declare test arguments
+  cryptotest_hmac_hash_alg_t uj_hash_alg;
+  cryptotest_hmac_key_t uj_key;
+  cryptotest_hmac_message_t uj_message;
+  // Deserialize test arguments from UART
+  TRY(ujson_deserialize_cryptotest_hmac_hash_alg_t(uj, &uj_hash_alg));
+  TRY(ujson_deserialize_cryptotest_hmac_key_t(uj, &uj_key));
+  TRY(ujson_deserialize_cryptotest_hmac_message_t(uj, &uj_message));
+
+  otcrypto_key_mode_t key_mode;
+  unsigned int tag_bytes;
+  switch (uj_hash_alg) {
+    case kCryptotestHmacHashAlgSha256:
+      key_mode = kOtcryptoKeyModeHmacSha256;
+      tag_bytes = kOtcryptoHmacTagBytesSha256;
+      break;
+    case kCryptotestHmacHashAlgSha384:
+      key_mode = kOtcryptoKeyModeHmacSha384;
+      tag_bytes = kOtcryptoHmacTagBytesSha384;
+      break;
+    case kCryptotestHmacHashAlgSha512:
+      key_mode = kOtcryptoKeyModeHmacSha512;
+      tag_bytes = kOtcryptoHmacTagBytesSha512;
+      break;
+    case kCryptotestHmacHashAlgSha3_256:
+      // key_mode = kOtcryptoKeyModeHmacSha3_256;
+      // tag_bytes = kOtcryptoHmacTagBytesSha3_256;
+      // break;
+      OT_FALLTHROUGH_INTENDED;
+    case kCryptotestHmacHashAlgSha3_384:
+      // key_mode = kOtcryptoKeyModeHmacSha3_384;
+      // tag_bytes = kOtcryptoHmacTagBytesSha3_384;
+      // break;
+      OT_FALLTHROUGH_INTENDED;
+    case kCryptotestHmacHashAlgSha3_512:
+      // key_mode = kOtcryptoKeyModeHmacSha3_512;
+      // tag_bytes = kOtcryptoHmacTagBytesSha3_512;
+      // break;
+      OT_FALLTHROUGH_INTENDED;
+    default:
+      LOG_ERROR("Unsupported HMAC key mode: %d", uj_hash_alg);
+      return INVALID_ARGUMENT();
+  }
+  // Build the key configuration
+  otcrypto_key_config_t config = {
+      .version = kOtcryptoLibVersion1,
+      .key_mode = key_mode,
+      .key_length = uj_key.key_len,
+      .hw_backed = kHardenedBoolFalse,
+      .security_level = kOtcryptoKeySecurityLevelLow,
+  };
+  // Create buffer to store key
+  uint32_t key_buf[MaxKeyWords];
+  memcpy(key_buf, uj_key.key, uj_key.key_len);
+  // Create keyblob
+  uint32_t keyblob[keyblob_num_words(config)];
+  // Create blinded key
+  TRY(keyblob_from_key_and_mask(key_buf, kTestMask, config, keyblob));
+  otcrypto_blinded_key_t key = {
+      .config = config,
+      .keyblob_length = sizeof(keyblob),
+      .keyblob = keyblob,
+  };
+
+  // Create input message
+  uint8_t msg_buf[MaxMessageBytes];
```

Comment:
I made a similar comment on a previous PR: this method of allocation seems dangerous to me. The `MaxMessageBytes` appears to be a hardcoded value that's simply the maximum message length that happens to be in the current test vectors, but if later another test vector got added that had a longer length then the test would develop a buffer overflow error, which will be annoying and painful to debug because buffer overflows always are. There's no reason in C why you can't just allocate this dynamically on the stack to avoid that problem:
```suggestion
  uint8_t msg_buf[uj_message.message_len];
```
The same goes for the key and the tag, and any other test harnesses that use this method of allocation. Since it's the second time I'm seeing this pattern, I'm worried that other tests have it too.

Reply:
Thought I had fixed all of these. Thanks for the catch.